samples/gen-go/client/client.go

```diff
@@ -1079,6 +1081,11 @@ func (c *WagClient) doGetBookByIDRequest(ctx context.Context, req *http.Request,
 		return nil, &output
 
 	default:
+		var body interface{}
+		bs, err := ioutil.ReadAll(resp.Body)
+		if err == nil && json.Unmarshal(bs, &body) == nil {
```

Comment:
This is the kind of change I was thinking of, but I'll note:
- it seems like `body` is unused? I'm not sure if we want to special case if the body is JSON vs if it isn't - if the response is an unexpected status code we don't want to make assumptions about the body structure.
- It would be good to include the unknown status code in the error.

I also am not sure either way if it makes sense to create a `models.UnknownResponseCode` model to encapsulate this, or just use `fmt.Errorf`/`errors.New`

Reply:
ah ok. I got the sense that part of the solution would be some kind of inference about whether the server had created a new response code, and that the presence of a JSON body would be indicative of that

I like the idea of a new error type so will noodle on that

Reply:
@taylor-sutton just made changes to use `models.UnknownResponse` let me know what you think